app/models/classification.rb

```diff
@@ -99,9 +99,16 @@ def ns
 
   def self.classify(obj, category_name, entry_name, is_request = true)
     cat = Classification.lookup_by_name(category_name, obj.region_id)
-    unless cat.nil?
-      ent = cat.find_entry_by_name(entry_name, obj.region_id)
-      ent.assign_entry_to(obj, is_request) unless ent.nil? || obj.is_tagged_with?(ent.to_tag, :ns => "none")
+    return " - FAILED. Tag category '#{category_name}' not found in region #{obj.region_id}" if cat.nil?
```

Comment:
This used to return `nil` in this case, but now returns a string.  Should we raise instead?

Reply:
> This used to return `nil` in this case, but now returns a string. Should we raise instead?

👍 raising error would be cleaner - we would not need changes on API side